op-node/rollup/driver/finalized.go

```diff
@@ -0,0 +1,33 @@
+package driver
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+type finalized struct {
+	derive.L1Fetcher
+	l1Finalized func() eth.L1BlockRef
+}
+
+func NewFinalized(l1Finalized func() eth.L1BlockRef, fetcher derive.L1Fetcher) *finalized {
+	return &finalized{L1Fetcher: fetcher, l1Finalized: l1Finalized}
+}
+
+func (f *finalized) L1BlockRefByNumber(ctx context.Context, num uint64) (eth.L1BlockRef, error) {
+	// Don't apply the finalized wrapper if l1Finalized is empty (as it is during the startup case before the l1State is initialized).
+	l1Finalized := f.l1Finalized()
+	if l1Finalized == (eth.L1BlockRef{}) {
+		return f.L1Fetcher.L1BlockRefByNumber(ctx, num)
+	}
+	if num == 0 || num <= l1Finalized.Number {
+		return f.L1Fetcher.L1BlockRefByNumber(ctx, num)
+	}
```

Comment:
```suggestion
	if l1Finalized == (eth.L1BlockRef{}) || num == 0 || num <= l1Finalized.Number {
		return f.L1Fetcher.L1BlockRefByNumber(ctx, num)
	}
```

Reply:
l1Finalized.Number could be negative? Why the `num == 0` if it could be covered by `num <= l1Finalized.Number`?

Reply:
This is also code from https://github.com/celo-org/optimism/pull/209, the `num == 0` is a small optimization and can be removed.

https://github.com/celo-org/optimism/blob/celo8/op-node/rollup/confdepth/conf_depth.go#L36